src/jvmMain/kotlin/com/sdercolin/vlabeler/video/VideoPlayer.kt

```diff
@@ -0,0 +1,111 @@
+// see https://github.com/JetBrains/compose-jb/pull/1738 for direct control of awt/swing
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package com.sdercolin.vlabeler.video
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.SideEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.awt.LocalLayerContainer
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.round
+import com.sdercolin.vlabeler.env.Log
+import com.sdercolin.vlabeler.env.isMacOS
+import uk.co.caprica.vlcj.factory.discovery.NativeDiscovery
+import uk.co.caprica.vlcj.player.base.MediaPlayer
+import uk.co.caprica.vlcj.player.component.CallbackMediaPlayerComponent
+import uk.co.caprica.vlcj.player.component.EmbeddedMediaPlayerComponent
+import uk.co.caprica.vlcj.player.component.InputEvents
+import java.awt.BorderLayout
+import javax.swing.JPanel
+
+class MiniVideo() {
+    var mediaPlayerComponent: Any? = null
```

Comment:
Please create an interface for them, instead of using `Any`.

e.g.
```kotlin
sealed interface MediaPlayerContainer {
   val mediaPlayer: MediaPlayer
   // other properties or functions you need for both types of components
}

class EmbeddedMediaPlayerContainer( ... ): MediaPlayerContainer {

  private val component: EmbeddedMediaPlayerComponent = ..

  override val mediaPlayer: MediaPlayer get() = component.mediaPlayer()
}

class CallbackMediaPlayerContainer( ... ): MediaPlayerContainer {
  ..
}
```

Reply:
I use generic type `T: Component` instead of `Any` so that `component` can be accessed without additional branching:
```suggestion
class MediaPlayerComponent<T: Component>(
    val component: T
)

var mediaPlayerComponent: MediaPlayerComponent<*>? = null
```
In `VideoCore.kt`, now we can ```add(videoState.videoPlayer.mediaPlayerComponent?.component)```
compared to previous ```add(if (isMacOS) ... else ...)```.